tests/python/test_household.py

```diff
@@ -0,0 +1,377 @@
+        mock_row.keys.return_value = SAMPLE_DB_ROW.keys()
+        mock_database.query().fetchone.return_value = mock_row
+
+        # Make request
+        response = rest_client.get("/us/household/1")
+        data = json.loads(response.data)
+
+        assert response.status_code == 200
+        assert data["status"] == "ok"
+        assert (
+            data["result"]["household_json"] == SAMPLE_HOUSEHOLD_DATA["data"]
+        )
+
+    def test_get_nonexistent_household(self, rest_client, mock_database):
+        """Test getting a non-existent household."""
+        mock_database.query().fetchone.return_value = None
+
+        response = rest_client.get("/us/household/999")
+        data = json.loads(response.data)
+
+        assert response.status_code == 404
+        assert data["status"] == "error"
+        assert "not found" in data["message"]
+
+    def test_get_household_invalid_id(self, rest_client):
+        """Test getting a household with invalid ID."""
+        response = rest_client.get("/us/household/invalid")
+
+        assert response.status_code == 400
+        assert b"Invalid household ID" in response.data
+
+
+class TestCreateHousehold:
+    def test_create_household_success(
+        self, rest_client, mock_database, mock_hash_object
+    ):
+        """Test successfully creating a new household."""
+        # Mock database responses
+        mock_row = MagicMock(spec=LegacyRow)
+        mock_row.__getitem__.side_effect = lambda x: {"id": 1}[x]
+        mock_database.query().fetchone.return_value = mock_row
+
+        response = rest_client.post(
+            "/us/household",
+            json=SAMPLE_HOUSEHOLD_DATA,
+            content_type="application/json",
+        )
+        data = json.loads(response.data)
+
+        assert response.status_code == 201
+        assert data["status"] == "ok"
+        assert data["result"]["household_id"] == 1
+
+    def test_create_household_invalid_payload(self, rest_client):
+        """Test creating a household with invalid payload."""
+        invalid_payload = {
+            "label": "Test",
+            # Missing required 'data' field
+        }
+
+        response = rest_client.post(
+            "/us/household",
+            json=invalid_payload,
+            content_type="application/json",
+        )
+
+        assert response.status_code == 400
+        assert b"Missing required keys" in response.data
+
+    def test_create_household_invalid_label(self, rest_client):
+        """Test creating a household with invalid label type."""
+        invalid_payload = {
+            "data": {},
+            "label": 123,  # Should be string or None
+        }
+
+        response = rest_client.post(
+            "/us/household",
+            json=invalid_payload,
+            content_type="application/json",
+        )
+
+        assert response.status_code == 400
+        assert b"Label must be a string or None" in response.data
+
+
+class TestUpdateHousehold:
```

Comment:
**discussion/non-blocking** I like the way you have organized these tests with a class per operation and clearly named tests.

Have you looked at [pytest-describe](https://github.com/pytest-dev/pytest-describe) or similar ways of removing some of the boilerplate/repetitive aspects of the test syntax (I haven't used that specifically, but I've used similar frameworks in javascript and ruby)

Reply:
I have not, but will